src/test/java/ai/spice/FlightQueryTest.java

```diff
@@ -113,4 +113,43 @@ public void testRefreshSpiceOSS() throws ExecutionException, InterruptedExceptio
             fail("Should not throw exception: " + e.getMessage());
         }
     }
+
+    public void testRefreshWithOptionsSpiceOSS() throws ExecutionException, InterruptedException {
+        try {
+            String sql = "SELECT tpep_pickup_datetime, total_amount, passenger_count from taxi_trips limit 20;";
+            SpiceClient spiceClient = SpiceClient.builder()
+                    .build();
+
+            FlightStream preRefreshRes = spiceClient.query(sql);
+
+            int preRefreshRows = 0;
+
+            while (preRefreshRes.next()) {
+                VectorSchemaRoot root = preRefreshRes.getRoot();
+                preRefreshRows += root.getRowCount();
+            }
+
+            assertEquals("Expected row count does not match", 20, preRefreshRows);
+
+            RefreshOptions opts = new RefreshOptions().withRefreshSql("SELECT * FROM taxi_trips limit 10");
+
+            spiceClient.refresh_dataset("taxi_trips", opts);
+
+            // wait a couple seconds to let refresh run
+            Thread.sleep(10000);
```

Comment:
10s timeout is too large IMO for such test/case
Can we also add small refreshJitter to see if it will be accepted.

Reply:
It takes about 6 seconds to refresh for me on my local laptop.

Reply:
Wow - then it is good 👍